homeassistant/components/arcam_fmj/config_flow.py

```diff
@@ -16,12 +19,79 @@ class ArcamFmjFlowHandler(config_entries.ConfigFlow):
     VERSION = 1
     CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
 
+    def _get_entry(self, host, port, uuid=None):
+        for entry in self._async_current_entries():
+            if (entry.data[CONF_HOST] == host and entry.data[CONF_PORT] == port) or (
+                uuid and entry.data.get(CONF_UUID) == uuid
+            ):
+                return entry
+
+    async def async_step_user(self, user_info):
+        """Handle a discovered device."""
+        errors = {}
+
+        if user_info is not None:
+            friendly_name = "{} ({})".format(DEFAULT_NAME, user_info[CONF_HOST])
+
+            from . import DEVICE_SCHEMA
+
+            config = DEVICE_SCHEMA(user_info)
+            return self.async_create_entry(title=friendly_name, data=config)
+
+        context = self.context  # pylint: disable=no-member
+        fields = {
+            vol.Required(CONF_HOST, default=context.get(CONF_HOST)): str,
+            vol.Required(
+                CONF_PORT, default=context.get(CONF_PORT) or DEFAULT_PORT
+            ): int,
+            vol.Optional(CONF_UUID, default=context.get(CONF_UUID)): str,
```

Comment:
Can't we just fetch the SSDP description which includes the UUID ?

Reply:
That is what I intend to do. However, the thing is you can also connect via uart (net to uart device). And if you do, there is no way to get the ssdp. I've been in contact with the manufacturer and there is no way to get unique I'd via the normal command connection.

The uart connection is more reliable (tcp stack is crap) and support direct turn on without external features. So I really want to support that.

Reply:
I've decided to add a support in the arcam_fmj lib to act uart->net gateway. That way it can behave like the actual device and support ssdp as well.